### PR TITLE
minHash wrapper

### PR DESCRIPTION
**Pull request by joachimwolff**

### CODE REVIEW

File: tools/minHash/minHash.xml [hunk 1]
```diff
@@ -0,0 +1,196 @@
+<tool id="minHashTool" name="MinHash" version="0.1">
+  <description>Approximate k-neighbors search for sparse datasets</description>
+  <requirements>
+    <requirement type="package" version="0.2.3">sparse-neighbors-search</requirement>
+    <requirement type="package" version="1.11.3">numpy</requirement>
```
Comment (bgruening) [inline tools/minHash/minHash.xml]: numpy is already a dependency of sparse-neighbors-search

File: tools/minHash/.shed.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+name: minHash
+owner: joachimwolff
+description: A tool to compute the nearest neighbors in a high dimensional sparse dataset.
+
+long_description: |
+  MinHash is an approximate nearest neighbors search algorithm for high dimensional sparse datasets. It computes first a candidate set with MinHash and after the 
+  exact neighbors with the euclidean distance or cosine similarity.
+
+categories:
+  - Machine Learning
```
Comment (bgruening) [inline tools/minHash/.shed.yml]: Is this at the TS?

File: tools/minHash/.shed.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  
+remote_repository_url: https://github.com/joachimwolff/galaxytools/tree/minHash/tools/minHash
```
Comment (bgruening) [inline tools/minHash/.shed.yml]: Can this be fixed?

File: tools/minHash/.shed.yml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+name: minHash
+owner: joachimwolff
```
Comment (bgruening) [inline tools/minHash/.shed.yml]: This needs to be a TS user.

File: tools/minHash/.shed.yml [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+long_description: |
+  MinHash is an approximate nearest neighbors search algorithm for high dimensional sparse datasets. It computes first a candidate set with MinHash and after the 
+  exact neighbors with the euclidean distance or cosine similarity.
+
+categories:
+  - Machine Learning
+  
+remote_repository_url: https://github.com/joachimwolff/galaxytools/tree/minHash/tools/minHash
+
+auto_tool_repositories:
```
Comment (bgruening) [inline tools/minHash/.shed.yml]: This is only needed if you have more than 1 tool in a repo that you want to demultiplex.

File: tools/minHash/minHash.xml [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+dist, indices = minHash.fit_kneighbors(data_matrix)
+mmwrite(open("$dist", 'w+'), dist) 
+mmwrite(open("$indices", 'w+'), indices)
+      ]]>
+    </configfile>
+  </configfiles>
+
+  
+    <inputs>
+        <param name="input_file" type="data" format="mtx" label="Source file" help="Needs to be a sparse matrix from scipy/numpy in csr format."/>
```
Comment (bgruening) [inline tools/minHash/minHash.xml]: What is mtx and is this a Galaxy datatype already?

Comment (joachimwolff) [inline tools/minHash/minHash.xml]: Mtx is a Matrix Market file which is offered to store and exchange matrix data. See https://en.wikipedia.org/wiki/Matrix_Market_exchange_formats or https://docs.scipy.org/doc/scipy/reference/generated/scipy.io.mmread.html#scipy.io.mmread

I tried it with [numpy's npy](https://docs.scipy.org/doc/numpy/reference/generated/numpy.load.html) too but in my experience it is unreliable. 

I don't know if this [list](https://wiki.galaxyproject.org/Learn/Datatypes) of datatypes is intended to be exhaustive, mtx is not listed there. Which means, no mtx is not a Galaxy datatype already.

File: tools/minHash/.shed.yml (hunk 4 above)
Comment (joachimwolff) [inline tools/minHash/.shed.yml]: Should I leave it blank, use your name...?

File: tools/minHash/minHash.xml (hunk 6 above)
Comment (bgruening) [inline tools/minHash/minHash.xml]: Use the source Luke :)
https://github.com/galaxyproject/galaxy/blob/dev/config/datatypes_conf.xml.sample
That it should be added I think.

File: tools/minHash/.shed.yml (hunk 2 above)
Comment (joachimwolff) [inline tools/minHash/.shed.yml]: Given this [list](https://toolshed.g2.bx.psu.edu/repository/browse_categories), the answer is no. It does not fit in any category listed there.

File: tools/minHash/.shed.yml (hunk 4 above)
Comment (bgruening) [inline tools/minHash/.shed.yml]: bgruening, rnateam ... something like that.

File: tools/minHash/.shed.yml (hunk 2 above)
Comment (bgruening) [inline tools/minHash/.shed.yml]: I ask for one.

File: tools/minHash/.shed.yml [hunk 7]
```diff
@@ -0,0 +1,11 @@
+name: minHash
+owner: bgruening
+description: A tool to compute the nearest neighbors in a high dimensional sparse dataset.
+
+long_description: |
+  MinHash is an approximate nearest neighbors search algorithm for high dimensional sparse datasets. It computes first a candidate set with MinHash and after the 
+  exact neighbors with the euclidean distance or cosine similarity.
+  
+categories:
+
```
Comment (bgruening) [inline tools/minHash/.shed.yml]: Categories ...

File: tools/minHash/minHash.xml [hunk 8]
```diff
@@ -0,0 +1,190 @@
+<tool id="minHashTool" name="MinHash" version="0.1">
+  <description>Approximate k-neighbors search for sparse datasets</description>
+  <requirements>
+    <requirement type="package" version="0.3">sparse-neighbors-search</requirement>
+  </requirements>
+  <command>
+    <![CDATA[
+    python "$minHash_script" '$inputs' "\${GALAXY_SLOTS:-8}"
```
Comment (bgruening) [inline tools/minHash/minHash.xml]: single quotes

**Pull request closed without merging**